### DISCUSSION TITLE
oMBOOT hack

### DISCUSSION THREAD

**Discussion by catch-point** [gopro/labs#118] [Show and tell]
Use `oMBOOT="!S!1R"` to repeat subscript in larger QR Code.

I find I use this hack all the time with my Hero10 Black and the QRControl app. It runs the quoted script once at the end of the code (not at boot the way `!MBOOT=""` does). The same thing could be done with `!SAVE`/`!L` but it would require multiple QR Codes since `!SAVE` must be at the end of the script.

`oMBOOT` allows me to Set Date and Time in the same QR Code that uses a repeat command in QRControl. For example I have QR Code that setups my camera for overnights photos (including Date and Time) and looks something like this `mPNdLoMBOOT="!108NQ!S!R"`

During a regatta, I'm not close enough to the camera to use a QR Code and the battery won't last long enough if it's on the whole time, so I have to program it before I get on the water. I use this code to have the camera turn off until close to my start time, then wait until the race begins using hindsight and the speed trigger. Repeat is needed in the event of a false start. I use something like this: `!10:15NQv1oMBOOT="mVhS2!SK15-10H5hS0!R"`

What other models does this hack work with? Anybody else do something similar?

**Comment by dnewman-gpsw** [comment]
Interesting hack, using oMBOOT="subroutine".  I should think of formalizing that.  Many of the camera models show support this.  I like it.

**Comment by dnewman-gpsw** [comment]
I added this useful hack to https://gopro.github.io/labs/control/extensions/

**Comment by fxstein** [comment]
Very interesting! I need to think through how this could work in combination with The Remote - we use it extensively to control multiple cameras per vehicle, but it's not always reliable or people forget to start and stop timelapse shoots.
